### PR TITLE
Optional tor static builds

### PR DESCRIPTION
**Pull request by offlinehacker**
I need this for portable tor client. This also add optional static builds for openssl and libevent.

### CODE REVIEW

**Comment by 7c6f434c** [comment]
@offlinehacker so, is this PR correct w.r.t. your plan? Should it be merged or closed or fixed?

**Comment by offlinehacker** [comment]
Well i left it open for so long as we don't have adapter for static builds.

File: pkgs/tools/security/tor/default.nix [hunk 1]
```diff
@@ -10,9 +12,17 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ libevent openssl zlib ];
```
Comment (vcunat) [inline pkgs/tools/security/tor/default.nix]: You always have non-static versions in ``buildInputs``. That seems dangerous.

**Comment by vcunat** [comment]
Maybe it would be better to solve this by a stdenv adapter which would also set ``dontDisableStatic``, ``--disable-dynamic`` and perhaps ``allowedReferences = [];``, etc. It should be doable by adding a hook to set such things. Then I imagine one might define version of whole ``pkgs`` (not evaluated by default) where the default stdenv would be replaced by this modified one.

**Pull request closed without merging**